src/Constraint/MultiConstraint.php

```diff
@@ -233,9 +233,9 @@ private static function optimizeConstraints(array $constraints, $conjunctive)
             for ($i = 1, $l = \count($constraints); $i < $l; $i++) {
                 $right = $constraints[$i];
                 if (
-                    $left instanceof MultiConstraint
+                    $left instanceof self
```

Comment:
Does this work properly on PHP 5.3?

Reply:
It's one of the most basic self usages which works since PHP 5.0 or why do you question it?

Reply:
Because PHP 5.4 added extra stuff for `self` that people thought always worked, but actually didn't.